skorch/tests/test_helper.py

```diff
@@ -280,3 +281,23 @@ def forward(self, X):
         assert isinstance(net.optimizer_, torch.optim.SGD)
         assert len(net.optimizer_.param_groups) == 1
         assert net.optimizer_.param_groups[0]['momentum'] == 0.9
+
+
+class TestPredefinedSplit():
+
+    @pytest.fixture
+    def predefined_split(self):
+        from skorch.helper import predefined_split
+        return predefined_split
+
+    def test_pickle(self, tmpdir, predefined_split, data):
+        from skorch.dataset import Dataset
+
+        valid_dataset = Dataset(*data)
+        train_split = predefined_split(valid_dataset)
+
+        tmp_fn = tmpdir.mkdir('predefined_split').join('test.pkl')
+
+        # does not raise
+        with tmp_fn.open('wb') as f:
+            pickle.dump(train_split, f)
```

Comment:
Maybe just dump the pickle into a string using `pickle.dumps`? Saves the temporary file hassle.